### PR TITLE
#29 Add in start of replacing debig enabling with various scenarios

### PR DESCRIPTION
**Pull request by labboy0276**

### CODE REVIEW

File: bin/hyperdrive [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+
 // Check for --debug and internally set DEBUG=* if its set
 // @TODO: we should make debug a string flag so it can work like setting DEBUG=something
 // @TODO: should we handle --debug on our own or use something lightweight like minimist?
 // @TODO: or maybe add this as a util?
 // @TODO: we should give preference to process.env.DEBUG if its set.
-if (process.argv.slice(2).find(element => element === '--debug') === '--debug') {
-  require('debug').enable('*'); // eslint-disable-line node/no-extraneous-require
+
+if (process.env.DEBUG !== undefined || argv.hasOption('--debug')) {
```
Comment (pirog) [inline bin/hyperdrive:11]: @labboy0276 this is good but i dont think it is exactly the logic we want. When `process.env.DEBUG` _has a value_ then we don't have to do anything. The `debug` module will know how to handle it. 

So we want to wrap our arg handling in something like `if (process.env.DEBUG === undefined || null || '') && argv.hasOption('--debug')` and then inside the conditional we only need to be concerned with setting `debug.enable` to the value of the `--debug` flag.

**Review by pirog — CHANGES_REQUESTED** [review]
See code comments. I know it's pedantic but let's also get into a good habit of putting the issue number in the commit message like `#29: some shit i did`. It's definitely if we have to look back and see what someone did.

File: bin/hyperdrive (hunk 1 above)
Comment (labboy0276) [reply bin/hyperdrive:11]: @pirog ok I did that if they pass in both the DEBUG env and the flag.  This way it overrides the flag

**Review by pirog — CHANGES_REQUESTED** [review]

File: test/commands/argv.test.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    });
+
+    it('hasOption should return true with --debug', () => {
+      const hasOption = argv.hasOption('--debug');
+      expect(hasOption).to.equal(true);
+    });
+
+    it('hasOption should return false with --trill', () => {
+      const hasOption = argv.hasOption('--trill');
+      expect(hasOption).to.equal(false);
```
[the remark this answers is not in the document]
Comment (labboy0276) [reply test/commands/argv.test.js:24]: @pirog see comment in slack

**Review by pirog — APPROVED** [review]

**Comment by pirog** [comment]
@labboy0276 this code is most righteous.

**Pull request merged**